pex/pex_builder.py

```diff
@@ -664,17 +672,34 @@ def build(
             else:
                 shutil.rmtree(tmp_pex, True)
 
+        if not compress:
+            if cache_dists is True:
+                pex_warnings.warn(
```

Comment:
So, say there is some legitimate use case, aka the user knows better. You've now pissed them off by nagging them. It's much cleaner if you just fail fast if that makes sense to do. I'm not commenting on whether or not there is a legitimate use case, my intent is just to point out the cost of combinatorial feature explosion and the nanny vs. document vs. trust the user tradeoffs.